### PR TITLE
Changes to new Proxy implementation

### PR DESCRIPTION
**Pull request by alexhernandezgarcia**

### CODE REVIEW

**Comment by alexhernandezgarcia** [comment]
Run of 5k iterations with flowmatch: [comet](https://www.comet.com/alexhernandezgarcia/gfn-grid/46ddc804a9fb4f3ea5bbf4fe3a34d991)

**Comment by nikita-0209** [comment]
haven't gone through the code yet but my first feeling is something is up with the rewards. I expected them to be in the range of 0.6, instead of 1.8. Also, I remember `kl` ideally reaches `1e-5` in 2k iterations. Will check today night.

**Comment by alexhernandezgarcia** [comment]
> 

Uhm right, that's odd. Good catch!

File: src/gflownet/envs/base.py [hunk 1]
```diff
@@ -36,9 +36,11 @@ def __init__(
         self.reward_func = reward_func
         self.energies_stats = energies_stats
         self.denorm_proxy = denorm_proxy
-        # TODO: remove oracle as not required in env I think
-        self.oracle = oracle_func
         self.proxy = proxy
+        if oracle is None:
```
Comment (nikita-0209) [inline src/gflownet/envs/base.py:40]: Why do we need a `oracle` in `class Gflownetenv`?

Comment (nikita-0209) [reply src/gflownet/envs/base.py:40]: Oh wait, just to confirm, this is for `true_density`?

File: src/gflownet/envs/grid.py [hunk 2]
```diff
@@ -64,24 +64,6 @@ def __init__(
...
-        if proxy:
-            self.proxy = proxy
-        else:
-            self.proxy = self.oracle
-        self.reward = (
-            lambda x: [0]
-            if not self.done
-            else self.proxy2reward(self.proxy(self.state2oracle(x)))
-        )
-        self._true_density = None
```
Comment (nikita-0209) [inline src/gflownet/envs/grid.py:83]: Not really imp, but is there any reason we removed the initialization `self._true_density` from the `init()`?

**Comment by nikita-0209** [comment]
Ah sorry my bad

> > 
> 
> ~Uhm right, that's odd. Good catch!~
> 
> The rewards are exp(-beta * proxy) so they can be 1.8. It's the proxy values that can be up to -0.6, which is the case.

Ah right okay, sorry my bad, thought `power` with `reward_beta=reward_norm=1` was being used

**Review by nikita-0209 — APPROVED** [review]

**Pull request merged**